### DISCUSSION TITLE
What message publisher should I use?

### DISCUSSION THREAD

**Discussion by Gigioliva** [googleapis/nodejs-pubsub#1890] [Q&A]
I see that we have two ways to publish a message:

- [topic.publishMessage](https://github.com/googleapis/nodejs-pubsub/blob/f12c644931c9c23613f966c561e0cd9acf6f8cb4/samples/publishMessage.js#L51)
- [v1.PublisherClient.publish](https://github.com/googleapis/nodejs-pubsub/blob/f12c644931c9c23613f966c561e0cd9acf6f8cb4/samples/publishWithRetrySettings.js#L103C44-L103C51)

Which one should I use? and what is the difference between the two?

**Comment by TomyCesaille** [comment]
https://cloud.google.com/pubsub/docs/retry-requests doc uses [v1.PublisherClient.publish](https://github.com/googleapis/nodejs-pubsub/blob/f12c644931c9c23613f966c561e0cd9acf6f8cb4/samples/publishWithRetrySettings.js#L103C44-L103C51) but we can also do it with [topic.publishMessage](https://github.com/googleapis/nodejs-pubsub/blob/f12c644931c9c23613f966c561e0cd9acf6f8cb4/samples/publishMessage.js#L51) too (`gaxOpts -> longrunning` option).
